src/chunkserver/copyset_service.cpp

```diff
@@ -134,6 +134,30 @@ void CopysetServiceImpl::CreateCopysetNode2(RpcController *controller,
     LOG(INFO) << "Create " << request->copysets().size() << " copysets success";
 }
 
+void CopysetServiceImpl::RemoveCopysetNode(RpcController* controller,
+                                           const CopysetRequest* request,
+                                           CopysetResponse* response,
+                                           Closure* done) {
+    LOG(INFO) << "Receive remove copyset node request";
+
+    brpc::ClosureGuard doneGuard(done);
+
+    auto poolId = request->logicpoolid();
+    auto copysetId = request->copysetid();
+    auto groupId = ToGroupIdString(poolId, copysetId);
+
+    if (!copysetNodeManager_->IsExist(poolId, copysetId)) {
+        response->set_status(COPYSET_OP_STATUS_COPYSET_NOTEXIST);
+        LOG(WARNING) << "Remove copyset node " << groupId  << " not exist";
+    } else if (!copysetNodeManager_->PurgeCopysetNodeData(poolId, copysetId)) {
```

Comment:
Should we check copyset status before purge its data？

Reply:
> Should we check copyset status before purge its data？

I think there is a bug in my implemention. actually, the copyset node will not exist in the `copysetNodeManager_ ` if its data has broken, because the `copysetNodeManager_` will ignore the data broken's copyset in the load phase when chunkserver start. So we will never delete the copyset node success.

Reply:
> Should we check copyset status before purge its data？

In the latest implementation, i will only delete the broken copyset.